### PR TITLE
Add headless commerce plugin tools package

### PR DESCRIPTION
**Pull request by teleaziz**
docs, example, usage coming next, the goal is to give headless backends an easy way to connect with Builder, this will add preview options for collections and products, ability to use products/collection in Builder sections and as targeting attributes as well.

### CODE REVIEW

**Comment by steve8708** [comment]
if this is commerce specific should this be called commerce-plugins-tools or is this so abstract its not ultimately commerce related?

**Comment by teleaziz** [comment]
This is mostly commerce specific, `plugin-tools` maybe too generic I agree, I'll update

**Review by steve8708 — APPROVED** [review]
this is awesome

**Review by Dkendal — APPROVED** [review]

File: packages/commerce-plugin-tools/src/examples/swell.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { registerCommercePlugin } from '../index';
+registerCommercePlugin(
+  {
+    name: 'Swell',
+    id: '@builder.io/ecome-swell',
```
Comment (steve8708) [inline packages/commerce-plugin-tools/src/examples/swell.ts:5]: ecome? maybe we can still to our convention of plugin-[name], like `@builder.io/plugin-swell`

File: plugins/swell/LICENSE [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Copyright 2020 Steve Sewell <steve@builder.io>
```
Comment (steve8708) [inline plugins/swell/LICENSE:1]: lets make this `Copyright 2021 Builder.io, Inc <steve@builder.io>`

File: plugins/swell/package.json [hunk 3]
```diff
@@ -0,0 +1,129 @@
+{
+  "name": "@builder.io/ecom-swell-is",
```
Comment (steve8708) [inline plugins/swell/package.json:2]: `@builder.io/plugin-swell`? even though their name is swell.is they just go by swell

**Review by steve8708 — APPROVED** [review]

**Pull request merged**